python/vespa/vespa/package.py

```diff
@@ -468,3 +490,160 @@ def deploy(self, disk_folder: str, container_memory: str = "4G"):
+                                  .format(self.tenant, self.application, instance, job, run, last))
+
+            for step, entries in update['log'].items():
+                for entry in entries:
+                    self.print_log_entry(step, entry)
+            last = update.get('lastId', last)
+
+            if update['active']:
+                sleep(1)
+            else:
+                status = update['status']
+                if   status == 'success':                    return
+                elif status == 'error':                      raise RuntimeError("Unexpected error during deployment; see log for details")
+                elif status == 'aborted':                    raise RuntimeError("Deployment was aborted, probably by a newer deployment")
+                elif status == 'outOfCapacity':              raise RuntimeError("No capacity left in zone; please contact the Vespa team")
+                elif status == 'deploymentFailed':           raise RuntimeError("Deployment failed; see log for details")
+                elif status == 'installationFailed':         raise RuntimeError("Installation failed; see Vespa log for details")
+                elif status == 'running':                    raise RuntimeError("Deployment not completed")
+                elif status == 'endpointCertificateTimeout': raise RuntimeError("Endpoint certificate not ready in time; please contact Vespa team")
+                elif status == 'testFailure':                raise RuntimeError("Unexpected status; tests are not run for manual deployments")
+                else:                                        raise RuntimeError("Unexpected status '" + status + "'")
+
+    @staticmethod
+    def print_log_entry(step: str, entry: dict):
+        timestamp = strftime('%H:%M:%S', gmtime(entry['at'] / 1e3))
+        message = entry['message'].replace('\n', '\n' + ' '*23)
+        if step != 'copyVespaLogs' or entry['type'] == 'error':
+            print('{:<7} [{}]  {}'.format(entry['type'].upper(), timestamp, message))
+
+    def deploy(self, instance: str, application_package: ApplicationPackage) -> Vespa:
+        """
+        Deploy the given application package as the given instance in the Vespa Cloud dev environment.
+
+        :param instance: Name of this instance of the application, in the Vespa Cloud.
+        :param application_package: ApplicationPackage to be deployed.
+
+        :return: a Vespa connection instance.
+        """
+
+        region = self.get_dev_region()
+        job = 'dev-' + region
+        run = self.start_deployment(instance, job, application_package)
+        self.follow_deployment(instance, job, run)
+        endpoint_url = self.get_endpoint(instance, region)
+        return Vespa(url = endpoint_url, cert = self.data_cert_file.name)
+
+    def close(self):
+        self.connection.close()
+        self.data_cert_file.close()
```

Comment:
This deletes the data key/cert _forever_. Consider allowing the user to specify the certificate file, for persistent storage, if it's meaningful to allow the user to keep talking to its endpoint outside of the python code, or if the `VespaCloud` variable can't be kept in scope.

Reply:
I believe that using
```
vespa_cloud = VespaCloud("vespa-team", "ms-marco", "/Users/tmartins/sample_application/tmartins.vespa-team.pem")
app = vespa_cloud.deploy('from-notebook', app_package)
# run as many commands as one needs
app.close()
```
solves the case where the `VespaCloud` variable can't be kept in scope.

Reply:
I think it is best to let them generate the certificate themselves if they want to talk to the end-point outside of the python code. As the intent of this library is to make experiments from a python environment.

Reply:
Yes, you'd need to move the key/cert stuff to the `deploy(...)` call, but that's possible.

Reply:
You should prefer `with ...:`, though, to ensure resources are closed when exceptions are thrown.

Reply:
No, I don't need to change anything. My code above is wrong and I meant `vespa_cloud.close()` instead of `app.close()`

I don't need to move anything to `.deploy()` as the `Vespa` instance knows the location of the cert.